src/App.tsx

```diff
@@ -1,93 +1,136 @@
+/* eslint-disable @typescript-eslint/no-unused-expressions */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { TodosFilter } from './components/TodosFilter';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { SortBy, Todo } from './types';
+
+let todos = localStorage.getItem('todosStorage');
+let todosFromLocalStorage: Todo[] = [];
+
+if (todos !== null) {
+  todosFromLocalStorage = JSON.parse(todos);
+  if (!Array.isArray(todosFromLocalStorage)) {
+    todosFromLocalStorage = [];
+  }
+}
```

Comment:
you can simplify something like this
```suggestion
const todosFromLocalStorage = localStorage.getItem('todosStorage');
const parsedTodos = todosFromLocalStorage ? JSON.parse(todosFromLocalStorage) : [];

const todosArray = Array.isArray(parsedTodos) ? parsedTodos : [];
```